agent/stats/engine.go

```diff
@@ -90,6 +98,8 @@ type DockerStatsEngine struct {
 	// tasksToDefinitions maps task arns to task definition name and family metadata objects.
 	tasksToDefinitions map[string]*taskDefinition
 	taskToTaskStats    map[string]*StatsTask
+	// dockerIdToStatusMessageSince maps docker ID to timestamp of status message sent to TACS
+	dockerIdToStatusMessageSince map[string]time.Time
```

Comment:
this map is not persisted to db so will be lost when agent restarts. would it make more sense to store the message as a new field of [Container])https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/container/container.go#L140) (e.g. "FirelensStatusMessageUnsafe") so this data can be preserved together with the container?

Reply:
Good call out on Agent restart scenario! 
`dockerIdToStatusMessageSince` just stores the timestamp of the message sent for each firelens v2 container and does not store the message itself.  Status message persists on the file path specified so if the Agent restarts and file exists it just resends and stores the timestamp of message sent in `dockerIdToStatusMessageSince`. 

We could still add `FirelensStatusMessageSinceUnsafe` field to the container struct to track the last time status message was sent to TACS but wanted to avoid adding more fields tto container struc that are specific to firelens v2 container.

Reply:
got it. considering agent restart should be rare and worst case is to send a status message that has been sent, i don't have strong preference around storing it or not